gwcli/gateway.py

```diff
@@ -414,7 +414,7 @@ def _set_controls(self, controls):
     def summary(self):
         return "Gateways: {}".format(len(self.gateway_group.children)), None
 
-    def ui_command_reconfigure(self, attribute, value):
+    def ui_command_reconfigure(self, attribute=None, value=None, **kwargs):
```

Comment:
Should ```attribute``` and ```value``` be removed? What happens if someone uses ```reconfigure attribute=cmdsn_depth value=128```?

Reply:
I left it for compat support in case people had scripts that were interacting with the command already:

If you do

reconfigure attribute=cmdsn_depth value=12

then it still works like before. I just detect that and stick it in the controls dict that gets passed around.

If you mix like this:

reconfigure attribute=cmdsn_depth value=12 cmdsn_depth=13

then we would end up overwriting the second one (13) with the first (12). I do not think that issue will come up a lot.


If you mix like this:

reconfigure attribute=cmdsn_depth value=12 immediate_data=no

then we would detect that and pass both cmdsn_depth and immediate data in the controls dict that gets passed around.

Same with the below calls too.

If we want to ignore compat it is ok with me and I can drop that code. I think the chances of users having a script that calls these commands is very unlikely. I was not sure how strict we were.

Reply:
Ah -- fair point. I'm fine leaving it then assuming it gets optionally added to the override dict (and prevents you from passing ```attribute= value=XYZ```).

Reply:
@mikechristie Running "reconfigure  rbd.lun0 max_data_area_mb= osd_op_timeout=10" results in an error of "expected integer for max_data_area_mb". It looks like it's taking the "osd_op_timeout=10" as the param